### ISSUE TITLE
error: Loading unprepared module

### ISSUE THREAD

**Issue by deckchairlabs** [denoland/deno#14347]
tl/dr minimal reproduction lives here https://github.com/deckchairlabs/deno-loading-unprepared-module-error

I'm receiving the following error `error: Loading unprepared module: file:///workspace/deno.json`, when trying to do the below:

**_Note removing the compilerOptions related to jsx, the code runs without issue._**

main.ts
```ts
import config from "./deno.json" assert { type: "json" };
console.log(config);
```

deno.json
```json
{
  "compilerOptions": {
    "jsx": "react-jsx",
    "jsxImportSource": "react"
  },
  "importMap": "./importMap.json"
}
```

importMap.json
```json
{
  "imports": {
    "react": "https://esm.sh/react@18",
    "react/jsx-runtime": "https://esm.sh/react@18/jsx-runtime.js"
  }
}
```

**Comment by nayeemrmn** [comment]
Yeah importing `deno.json` is broken and will be hard to fix because of the `deno_graph` abstraction which puts config files and modules in the same specifier pool.

**Comment by deckchairlabs** [comment]
Oh I didn't realise this was related to importing the demo config. I'll check to see if it's the same issue for a different json file, and if not, we can close this.

**Comment by bartlomieju** [comment]
> Yeah importing `deno.json` is broken and will be hard to fix because of the `deno_graph` abstraction which puts config files and modules in the same specifier pool.

Are you saying that `deno_graph` is not able to load JSON module of `deno.json` if it was used as a config file? That seems like a bug that needs to be fixed.

**Comment by deckchairlabs** [comment]
Why does this only happen when trying to use the jsx-runtime compiler options? Does seem a little odd
